rplugin/python3/denite/ui/action.py

```diff
@@ -134,6 +136,61 @@ def _change_path(prompt, params):
     return prompt.denite.restart()
 
 
+def _change_word(prompt, params):
+    # NOTE: Respect the behavior of 'w' in Normal mode.
+    if prompt.caret.locus == prompt.caret.tail:
+        return prompt.denite.change_mode('insert')
+    pattern_set = build_keyword_pattern_set(prompt.nvim)
+    pattern = re.compile(
+        r'^(?:%s+|%s+|[^\s\x20-\xff]+|)\s*' % pattern_set
+    )
+    forward_text = pattern.sub('', prompt.caret.get_forward_text())
+    prompt.text = ''.join([
+        prompt.caret.get_backward_text(),
+        forward_text
+    ])
+    prompt.denite.change_mode('insert')
+
+
+def _change_line(prompt, params):
+    prompt.text = ''
+    prompt.caret.locus = 0
+    prompt.denite.change_mode('insert')
+
+
+def _move_caret_to_next_word(prompt, params):
+    pattern = re.compile('^\S+')
+    original_text = prompt.caret.get_forward_text()
+    substituted_text = pattern.sub('', original_text)
+    prompt.caret.locus += 2 + len(original_text) - len(substituted_text)
```

Comment:
I updated the regex to `^\S+\s+\S` so the magic number is no longer necessary (and it handles multiple white space character before the next word)